custom_components/homewhiz/climate.py

```diff
@@ -0,0 +1,260 @@
+import logging
+
+from dacite import from_dict
+from homeassistant.components.climate import (
+    FAN_AUTO,
+    FAN_HIGH,
+    FAN_LOW,
+    FAN_MEDIUM,
+    ClimateEntity,
+    ClimateEntityFeature,
+    HVACMode,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import STATE_UNAVAILABLE, TEMP_CELSIUS
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity import DeviceInfo
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from .api import ApplianceContents, ApplianceInfo, IdExchangeResponse
+from .config_flow import EntryData
+from .const import DOMAIN
+from .homewhiz import HomewhizCoordinator, appliance_type_by_code, brand_name_by_code
+
+_LOGGER: logging.Logger = logging.getLogger(__package__)
+
+program_dict: dict[str, HVACMode] = {
+    "AIR_CONDITIONER_MODE_COOLING": HVACMode.COOL,
+    "AIR_CONDITIONER_MODE_AUTO": HVACMode.AUTO,
+    "AIR_CONDITIONER_MODE_DRY": HVACMode.DRY,
+    "AIR_CONDITIONER_MODE_HEATING": HVACMode.HEAT,
+    "AIR_CONDITIONER_MODE_FAN": HVACMode.FAN_ONLY,
+}
+
+wind_strength_dict: dict[str, str] = {
+    "WIND_STRENGTH_LOW": FAN_LOW,
+    "WIND_STRENGTH_MID": FAN_MEDIUM,
+    "WIND_STRENGTH_HIGH": FAN_HIGH,
+    "WIND_STRENGTH_AUTO": FAN_AUTO,
+}
+
+
+def clamp(value: int):
+    return value if value < 128 else value - 128
```

Comment:
This is in [sensor.py](https://github.com/rowysock/home-assistant-HomeWhiz/blob/main/custom_components/homewhiz/sensor.py#L36) too. Maybe we can move it to a helper.py file?

Reply:
Done